lib/rack/cors.rb

```diff
@@ -14,17 +14,17 @@ def initialize(app, opts={}, &block)
       @app = app
       @debug_mode = !!opts[:debug]
 
-      if logger = opts[:logger]
-        if logger.respond_to? :call
+      if opts[:logger]
+        if opts[:logger].respond_to? :call
           @logger_proc = opts[:logger]
         else
-          @logger = logger
+          @logger = opts[:logger]
         end
       end
 
       if block_given?
         if block.arity == 1
-          block.call(self)
+          yield self
```

Comment:
To actually get a memory/performance benefit from this change you also need to remove the `&block` in the definition of this function.